daemon/graphdriver/overlay/overlay.go

```diff
@@ -122,20 +127,11 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		backingFs = fsName
 	}
 
-	// check if they are running over btrfs, aufs, zfs or overlay
-	switch fsMagic {
-	case graphdriver.FsMagicBtrfs:
-		logrus.Error("'overlay' is not supported over btrfs.")
-		return nil, graphdriver.ErrIncompatibleFS
-	case graphdriver.FsMagicAufs:
-		logrus.Error("'overlay' is not supported over aufs.")
-		return nil, graphdriver.ErrIncompatibleFS
-	case graphdriver.FsMagicZfs:
-		logrus.Error("'overlay' is not supported over zfs.")
-		return nil, graphdriver.ErrIncompatibleFS
-	case graphdriver.FsMagicOverlay:
-		logrus.Error("'overlay' is not supported over overlay.")
-		return nil, graphdriver.ErrIncompatibleFS
```

Comment:
the logrus.Error can be put into a defer func just before the switch statement and executed if err != nil (just name err something like errNotSupported and do the check in the defer). Then you can compact the switch cases in one line and return once (and print the correct FsMagic in the defer in logrus.Error)